cocos/physics-2d/box2d-wasm/physics-world.ts

```diff
@@ -0,0 +1,441 @@
+import { PhysicsAABBQueryCallback } from './platform/physics-aabb-query-callback';
+import { PhysicsRayCastCallback } from './platform/physics-ray-cast-callback';
+import { PhysicsContact } from './physics-contact';
+import { Contact2DType, Collider2D, RaycastResult2D } from '../framework';
+import { B2Shape2D } from './shapes/shape-2d';
+import { PhysicsDebugDraw } from './platform/physics-debug-draw';
+import { Node, find, Layers } from '../../scene-graph';
+import { director } from '../../game';
+
+const tempVec3 = new Vec3();
+const tempVec2_1 = new Vec2();
+const tempVec2_2 = new Vec2();
+const tempB2Vec2_1 = { x: 0, y: 0 };
+
+// const tempB2AABB = null;//new B2.AABB();
+
+const testResults: Collider2D[] = [];
+
+export class B2PhysicsWorld implements IPhysicsWorld {
+    protected _world: B2.World;
+    protected _bodies: B2RigidBody2D[] = [];
+    protected _animatedBodies: B2RigidBody2D[] = [];
+    protected _rotationAxis: Vec3 = new Vec3();
+    protected _physicsGroundBody: B2.Body;
+
+    protected _contactListener: B2.ContactListener;
+    protected _aabbQueryCallback: B2.QueryCallback;
+    protected _raycastQueryCallback: B2.RayCastCallback;
+
+    private _temoBodyDef: B2.BodyDef;
+    private _tempB2AABB: B2.AABB;
+
+    get impl (): B2.World {
+        return this._world;
+    }
+
+    get groundBodyImpl (): B2.Body {
+        return this._physicsGroundBody;
+    }
+
+    constructor () {
+        this._world = new B2.World({ x: 0, y: -10 });
+        this._physicsGroundBody = this._world.CreateBody(new B2.BodyDef() as B2.BodyDef);
+        PhysicsContactListener._BeginContact = this._onBeginContact;
+        PhysicsContactListener._EndContact = this._onEndContact;
+        PhysicsContactListener._PreSolve = this._onPreSolve;
+        PhysicsContactListener._PostSolve = this._onPostSolve;
+        this._contactListener = B2.ContactListener.implement(PhysicsContactListener.callback);
+        this._world.SetContactListener(this._contactListener);
+
+        this._aabbQueryCallback = B2.QueryCallback.implement(PhysicsAABBQueryCallback.callback);
+        this._raycastQueryCallback = B2.RayCastCallback.implement(PhysicsRayCastCallback.callback);
+
+        this._temoBodyDef = new B2.BodyDef();
+        this._tempB2AABB = new B2.AABB();
+    }
+
+    _debugGraphics: Graphics | null = null;
+    _b2DebugDrawer: B2.Draw | null = null;
+
+    _debugDrawFlags = 0;
+    get debugDrawFlags (): number {
+        return this._debugDrawFlags;
+    }
+    set debugDrawFlags (v) {
+        if (EDITOR_NOT_IN_PREVIEW) return;
+
+        if (!v) {
+            if (this._debugGraphics) {
+                this._debugGraphics.node.parent = null;
+            }
+        }
+
+        this._debugDrawFlags = v;
+    }
+
+    _checkDebugDrawValid (): void {
+        if (EDITOR_NOT_IN_PREVIEW) return;
+        if (!this._debugGraphics || !this._debugGraphics.isValid) {
+            let canvas = find('Canvas');
```

Comment:
How about using debug renderer to do this.

Reply:
Sure, I will try to use debug renderer in another PR.